### PR TITLE
Package Initialization

### PR DESCRIPTION
**Pull request by martinmaly**

### CODE REVIEW

File: porch/engine/pkg/engine/init.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	b, err := yaml.MarshalWithOptions(kptfile, &yaml.EncoderOptions{SeqIndent: yaml.WideSequenceStyle})
+	if err != nil {
+		return repository.PackageResources{}, nil, fmt.Errorf("failed to serialize Kptfile: %w", err)
+	}
+
+	if resources.Contents == nil {
+		resources.Contents = map[string]string{}
+	}
+
+	path := path.Join(m.spec.Subpackage, kptfilev1.KptFileName)
```
Comment (justinsb) [inline porch/engine/pkg/engine/init.go:64]: Nit: Shadowing `path` will give confusing results.  I try to do something like `kptfilePath := path.Join` or just `p := path.Join`

File: porch/engine/pkg/engine/engine.go [hunk 2]
```diff
@@ -102,6 +115,14 @@ func (cad *cadEngine) CreatePackageRevision(ctx context.Context, repositoryObj *
 
 func (cad *cadEngine) mapTaskToMutation(ctx context.Context, obj *api.PackageRevision, task *api.Task) (mutation, error) {
 	switch task.Type {
+	case api.TaskTypeInit:
```
Comment (justinsb) [inline porch/engine/pkg/engine/engine.go:118]: I do wonder if an init could be a clone with no upstream, but you're right that this is likely clearer!

**Review by justinsb — APPROVED** [review]
LGTM!

File: porch/engine/pkg/engine/engine.go (hunk 2 above)
Comment (martinmaly) [reply porch/engine/pkg/engine/engine.go:118]: We could also do clone from a 'singleton' (similar to what Docker does with `FROM scratch`.

The growing problem is representing actions as tasks, especially as Porch does more things automatically, things that there are no tasks for.

**Pull request merged**